.circleci/config.yml

```diff
@@ -25,19 +22,85 @@ jobs:
       - save_cache:
           paths:
             - node_modules
-          key: v1-dependencies-{{ checksum "package.json" }}
-        
+          key: v1-dependencies-v6-{{ checksum "package.json" }}
+
       - run:
           name: unit tests
           command: npm test
-          
+
       - run:
           name: coverage reporting
           command: npm run coverage:submit
 
-      - deploy:
-          name: publish
-          command: |
-            if [ "${CIRCLE_BRANCH}" == "master" ]; then
+      - persist_to_workspace:
+          root: ~/
+          paths:
+            - middy/*
+
+  build_next:
+    docker:
+      - image: circleci/node:8.10
+
+    working_directory: ~/middy
+
+    # These are almost the same set of steps as above.
+    # Yaml doesn't allow you to concatenate lists: https://github.com/yaml/yaml/issues/35
+    # Not in these steps:
+    #  - uploading code coverage
+    #  - saving workspace for the deploy job
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+          - v1-dependencies-v8-{{ checksum "package.json" }}
+          - v1-dependencies-v8-
+          - v1-dependencies-
+
+      - run:
+          name: install dependencies
+          command: npm install
+
+      - save_cache:
+          paths:
+            - node_modules
+          key: v1-dependencies-v8-{{ checksum "package.json" }}
+
+      - run:
+          name: unit tests
+          command: npm test
+
+  deploy:
+    docker:
+      - image: circleci/node:6.10
+
+    steps:
+        - attach_workspace:
+            at: ~/
+
+        - run:
+            name: configure NPM registry
+            command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> ~/.npmrc
+
```

Comment:
I am not really sure about this. We use the authentication only to publish the package (so it will be useful only in the workflow part). I'll have to see if this needs to be moved to the workflow step just before deployment.